### PR TITLE
limit staking

### PR DESCRIPTION
**Pull request by faboweb**
Optimizes queries. But takes the same time as the simple Mario version if you

### CODE REVIEW

File: src/signing/networkMessages/polkadot-testnet.js [hunk 1]
```diff
@@ -42,28 +42,68 @@ export async function StakeTx(senderAddress, { to, amount }, network) {
...
+      allClaimingTxs.push(api.tx.staking.payoutStakers(validators[0], era))
     })
-  }
+  console.timeEnd("rewards")
   console.log(allClaimingTxs)
   if (allClaimingTxs.length === 0) {
     throw new Error("There are no claimable rewards")
   }
-  return await getSignMessage(senderAddress, allClaimingTxs)
+  return await getSignMessage(senderAddress, allClaimingTxs[0])
```
Comment (faboweb) [inline src/signing/networkMessages/polkadot-testnet.js:106]:
```suggestion
  return await getSignMessage(senderAddress, allClaimingTxs)
```

**Comment by Bitcoinera** [comment]
> Optimizes queries. But takes the same time as the simple Mario version if you

This was cut out? What did you want to say here?

Always so cryptic 😎

File: src/signing/networkMessages/polkadot-testnet.js (hunk 1 above)
Comment (Bitcoinera) [reply src/signing/networkMessages/polkadot-testnet.js:106]: ???????????????

**Comment by faboweb** [comment]
Optimizes the query to be more performant. Reduces the amount of queries against the polkadot API. Maybe not needed. If we allow the user to query for `stakerRewards` why did we then store the rewards in the DB in the first place. Open for discussion.

File: src/signing/networkMessages/polkadot-testnet.js (hunk 1 above)
Comment (faboweb) [reply src/signing/networkMessages/polkadot-testnet.js:106]: that was just to test if one single tx works

**Comment by Bitcoinera** [comment]
> Optimizes the query to be more performant. Reduces the amount of queries against the polkadot API. Maybe not needed. If we allow the user to query for `stakerRewards` why did we then store the rewards in the DB in the first place. Open for discussion.

I 100% agree. DB is the way to go for Polkadot, since it itself is not so performant

**Comment by jbibla** [comment]
> Always so cryptic

@faboweb so true

**Comment by faboweb** [comment]
Next workshop: How Fabo can write so the team understands him. :D

**Comment by Bitcoinera** [comment]
> Next workshop: How Fabo can write so the team understands him. :D

Yes please!

**Pull request closed without merging**